upgradeable_contract/old_contract/src/test.rs

```diff
@@ -44,3 +44,20 @@ fn test() {
     let client = new_contract::Client::new(&env, &contract_id);
     assert_eq!(1010101, client.new_v2_fn());
 }
+
+#[test]
+#[should_panic(expected = "HostError: Error(Contract, #1)")]
```

Comment:
Could we use `try_init` and an assertion on the returned error instead of panic? The current setup is kind of brittle, e.g. it allows first `init` call to panic.

Reply:
Sure. Note however I followed the examples with `try_...` and I'm getting the following compilation error:

```
error[E0308]: mismatched types
  --> src/test.rs:61:41
   |
61 |     assert_eq!(client.try_init(&admin), Err(Ok(Error::AlreadyInitialized)),);
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Result<Result<(), ConversionError>, Result<..., ...>>`, found `Result<_, Result<Error, _>>`
```

The test looks like this now:
```
fn test_cannot_re_init() {
    let env = Env::default();
    env.mock_all_auths();

    // Note that we use register_contract_wasm instead of register_contract
    // because the old contracts WASM is expected to exist in storage.
    let contract_id = env.register_contract_wasm(None, old_contract::WASM);

    let client = old_contract::Client::new(&env, &contract_id);
    let admin = Address::generate(&env);
    client.init(&admin);
    assert_eq!(client.try_init(&admin), Err(Ok(Error::AlreadyInitialized)));
}
```

And I've imported the error with `use super::*;`

Any idea of what could be wrong? It sounds like a problem with the Err(Ok()) wrappers?

Reply:
The expected result looks correct to me (the inner result for the error case can be error itself in case if your contract panics without returning your error). Could you maybe commit this change, so that we could patch and debug?